Core/Object Arts/Dolphin/System/Win32/HTTP_LOGGING_INFO.cls

```diff
@@ -44,7 +44,9 @@ DirectoryName
 DirectoryName: anUtf16String
 	"Set the receiver's 'DirectoryName' field to the value of the argument, anUtf16String"
 
-	bytes uintPtrAtOffset: ##(self offsetOf: #DirectoryName) put: anUtf16String yourAddress!
+	directoryName := anUtf16String asUtf16String.
```

Comment:
These are auto-generated methods. They'll get overwritten the next time the structure is compiled out. You could just take them out of the auto-generated category to prevent that, but the normal thing to do would be to add a wrapper method (in this case it would be `directoryName:`) that calls the auto-generated method and adds the extra to perform the conversion and save the string to prevent it being GC'd.
The setting of the length could also be generated automatically if the IDL is better defined. I updated it (see [blairmcg/winhttpservertlb](url) https://github.com/blairmcg/winhttpservertlb) and re-generated this and I got:

```
DirectoryName: anUtf16String
	"Set the receiver's 'DirectoryName' field to the value of the argument, anUtf16String"

	bytes uintPtrAtOffset: ##(self offsetOf: #DirectoryName) put: anUtf16String yourAddress.
	self DirectoryNameLength: anUtf16String size << 1
```
It also regenerates the reader as:
```
DirectoryName
	"Answer the <Utf16String> value of the receiver's 'DirectoryName' field."

	^Utf16String fromAddress: (bytes intPtrAtOffset: ##(self offsetOf: #DirectoryName))
		length: self DirectoryNameLength >> 1
```
The original generated reader is assuming it is a null-terminated string, which is incorrect since from the structure definition it has a counted length and the null terminator cannot be assumed. Very often there will be one due to allocation granularity, but that is just luck really.

If you find other issues with the structure definitions it would be good to try and update the original IDL used to generate them; there is quite a lot of richness that can be expressed using extension attributes including custom ones understood by Dolphin that I added to capture extra metada.

However, since you are the person mainly working with this I'll leave it up to you what you want to do and commit this anyway.

Reply:
Ok, I see the .tlb is there in appveyor just built. Think I don't need any more.

Reply:
@blairmcg generating the wrapper using your last tlb shows changes in several files not related to the changes discussed here:

![image](https://user-images.githubusercontent.com/2080171/43051267-aebc46a2-8ded-11e8-832e-f500669ea862.png)

Mostly changes boolean parameters for integer and void* by PVOID.

It's ok? Should I commit all this files?

Reply:
@fxgallego, that seems to be a regression. Let me check out why that is.